docs/learn.md

```diff
@@ -721,20 +723,1250 @@ template](https://github.com/melange-re/melange-opam-template).
+where the resulting JavaScript values can be directly inlined at the call sites.
+
+#### Special identity external
+
+One external worth mentioning is the following one:
+
+```ocaml
+type foo = string
+type bar = int
+external danger_zone : foo -> bar = "%identity"
+```
+
+This is a final escape hatch which does nothing but convert from the type `foo`
+to `bar`. In the following sections, if you ever fail to write an `external`,
+you can fall back to using this one. But try not to.
+
+Let’s now see all the ways to use JavaScript from Melange.
+
+### Generate raw JavaScript
+
+It is possible to directly write JavaScript code from a Melange file. This is
+unsafe, but it can be useful for prototyping or as a escape hatch.
+
+To do it, we will use the `bs.raw`
+[extension](https://v2.ocaml.org/manual/extensionnodes.html):
+
+```ocaml
+let add = [%bs.raw {|
+  function(a, b) {
+    console.log("hello from raw JavaScript!");
+    return a + b;
+  }
+|}]
+
+let () = Js.log (add 1 2)
+```
+
+The `{||}` strings are called ["quoted
+strings"](https://ocaml.org/manual/lex.html#quoted-string-id). They are similar
+to JavaScript’s template literals, in the sense that they are multi-line, and
+there is no need to escape characters inside the string.
+
+Using one `%` (`[%bs.raw <string>]`) is useful to define expressions (function
+bodies, or other values) where the implementation is directly JavaScript. This
+is useful as we can attach the type signature already in the same line, to make
+our code safer. For example:
+
+```ocaml
+[%%bs.raw "var a = 1"]
+
+let f : unit -> int = [%bs.raw "function() {return 1}"]
+```
+
+Using two percentages `%%` (`[%%bs.raw <string>]`) is reserved for top-level
+definitions, for example:
+
+```ocaml
+[%%bs.raw "var a = 1"]
+```
+
+### Debugger
+
+Melange allows to inject a `[%bs.debugger]` expression:
+
+```ocaml
+let f x y =
+  [%bs.debugger];
+  x + y
+```
+
+Output:
+
+```javascript
+function f (x,y) {
+  debugger; // JavaScript developer tools will set an breakpoint and stop here
+  x + y;
+}
+```
+
+### Detect global variables
+
+Melange provides a relatively type safe approach to use globals that might be
+defined either in the JavaScript runtime environment: `bs.external`.
+
+`[%bs.external id]` will check if the JavaScript value `id` is `undefined` or
+not, and return an `Option.t` value accordingly.
+
+For example:
+
+```ocaml
+let () = match [%bs.external __DEV__] with
+| Some _ -> Js.log "dev mode"
+| None -> Js.log "production mode"
+```
+
+Another example:
+
+```ocaml
+let () = match [%bs.external __filename] with
+| Some f -> Js.log f
+| None -> Js.log "non-node environment"
+```
+
+### Bind to JavaScript objects
+
+JavaScript objects are used in a variety of use cases:
+
+- As a fixed shape "record" or "struct" in other languages (like OCaml and C).
+- As a map or dictionary.
+- As a class.
+- As a module to import/export.
+
+Melange cleanly separates the binding methods for JavaScript objects based on
```

Comment:
cleanly seems unnecessary (compared to what?)
```suggestion
Melange separates the binding methods for JavaScript objects based on
```

Reply:
Agree, in general I have been trying to remove as many adjectives as possible, and stick to purely descriptive language.

Removed in https://github.com/melange-re/documentation-site/pull/12/commits/3fb7df5d8288cee48739f79312e06acaf6a5e048.